### PR TITLE
Add controller/worker multi-machine setup

### PR DESCRIPTION
**Pull request by JayH5**

### CODE REVIEW

**Comment by jerith** [comment]
Are the upstream puppet module changes something we could split out into a separate PR?

File: puppet/run-tests.sh [hunk 1]
```diff
@@ -12,6 +12,7 @@ puppet-catalog-test -v \
     -M manifests/default.pp \
     --fact osfamily=Debian \
     --fact ipaddress_eth0=10.2.3.4 \
+    --fact ipaddress_eth1=192.168.0.2 \
```
Comment (jerith) [inline puppet/run-tests.sh]: What do we need this second IP for?

**Comment by JayH5** [comment]
Not really. The older versions of the seed_stack module were broken for multi-machine setups.

**Comment by jerith** [comment]
I meant split it out into a PR that lands before this one.

File: puppet/run-tests.sh (hunk 1 above)
Comment (JayH5) [inline puppet/run-tests.sh]: Oh I was using this before.. can remove now

File: puppet/README.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Puppet provisioning
 The Vagrant box is provisioned using [Puppet](http://docs.puppetlabs.com/puppet/3/reference/).
 
-The primary module used to provision the box is [`praekeltfoundation/seed_stack`](https://github.com/praekeltfoundation/puppet-seed_stack). See the documentation for the module for more information.
+The primary module used to provision the box is [`praekeltfoundation/seed_stack`](https://forge.puppetlabs.com/praekeltfoundation/seed_stack). See the documentation for the module for more information.
```
Comment (JayH5) [inline puppet/README.md:4]: Oops this should have been in the other PR.. oh well...

**Comment by jerith** [comment]
:+1:

**Pull request merged**